homeassistant/components/apple_tv/__init__.py

```diff
@@ -70,205 +70,257 @@ def ensure_list(value: Union[T, Sequence[T]]) -> Sequence[T]:
     extra=vol.ALLOW_EXTRA,
 )
 
-# Currently no attributes but it might change later
-APPLE_TV_SCAN_SCHEMA = vol.Schema({})
 
-APPLE_TV_AUTHENTICATE_SCHEMA = vol.Schema({ATTR_ENTITY_ID: cv.entity_ids})
+async def async_setup(hass, config):
+    """Set up the Apple TV integration."""
+    if DOMAIN not in config:
+        return True
+
+    for conf in config[DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=conf,
+            )
+        )
 
+    return True
 
-def request_configuration(hass, config, atv, credentials):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
 
-    async def configuration_callback(callback_data):
-        """Handle the submitted configuration."""
+async def async_setup_entry(hass, entry):
+    """Set up a config entry for Apple TV."""
+    identifier = entry.data[CONF_IDENTIFIER]
+    manager = AppleTVManager(hass, entry)
+    hass.data.setdefault(DOMAIN, {})[identifier] = manager
 
-        pin = callback_data.get("pin")
+    @callback
+    def on_hass_stop(event):
+        """Stop push updates when hass stops."""
+        asyncio.ensure_future(manager.disconnect(), loop=hass.loop)
 
-        try:
-            await atv.airplay.finish_authentication(pin)
-            hass.components.persistent_notification.async_create(
-                "Authentication succeeded!<br /><br />Add the following "
-                "to credentials: in your apple_tv configuration:<br /><br />"
-                "{0}".format(credentials),
-                title=NOTIFICATION_AUTH_TITLE,
-                notification_id=NOTIFICATION_AUTH_ID,
-            )
-        except DeviceAuthenticationError as ex:
-            hass.components.persistent_notification.async_create(
-                "Authentication failed! Did you enter correct PIN?<br /><br />"
-                "Details: {0}".format(ex),
-                title=NOTIFICATION_AUTH_TITLE,
-                notification_id=NOTIFICATION_AUTH_ID,
-            )
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, on_hass_stop)
 
-        hass.async_add_job(configurator.request_done, instance)
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "media_player")
+    )
 
-    instance = configurator.request_config(
-        "Apple TV Authentication",
-        configuration_callback,
-        description="Please enter PIN code shown on screen.",
-        submit_caption="Confirm",
-        fields=[{"id": "pin", "name": "PIN Code", "type": "password"}],
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "remote")
```

Comment:
Use for loop to iterate over platform strings

Reply:
Sure!